src/main/java/org/zalando/nakadi/service/timeline/LocalLocking.java

```diff
@@ -70,4 +45,34 @@ public Closeable workWithEventType(final String eventType, final long timeoutMs)
             }
         };
     }
+
+    public Set<String> getUnlockedEventTypes(final Set<String> lockedEventTypes) {
+        final Set<String> unlockedEventTypes = new HashSet<>();
+        synchronized (lock) {
+            for (final String item : this.lockedEventTypes) {
+                if (!lockedEventTypes.contains(item)) {
+                    unlockedEventTypes.add(item);
+                }
+            }
+        }
+        return unlockedEventTypes;
+    }
```

Comment:
Isn't it simplier 
```
synchronized (lock) {
    return this.lockedEventTypes.stream()
            .filter(v -> !lockedEventTypes.contains(v))
            .collect(Collectors.toSet));
}
```

Reply:
👍 done here https://github.com/zalando/nakadi/pull/796/commits/0c2f80fa08a4b8087474b97673f34fa13ffef7b3